lis3dh/lis3dh.go

```diff
@@ -0,0 +1,133 @@
+// Package lis3dh provides a driver for the LIS3DH digital accelerometer.
+//
+// Datasheet:
+package lis3dh
+
+import (
+	"bytes"
+	"encoding/binary"
+	"machine"
+	"time"
+)
+
+// Device wraps an I2C connection to a LIS3DH device.
+type Device struct {
+	bus     machine.I2C
+	Address uint16
+	x, y, z int16
+}
+
+// New creates a new LIS3DH connection. The I2C bus must already be configured.
+//
+// This function only creates the Device object, it does not touch the device.
+func New(bus machine.I2C) Device {
+	return Device{bus: bus, Address: Address0}
+}
+
+// Configure sets up the device for communication
+func (d *Device) Configure() {
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL5, []byte{0x80})
+	time.Sleep(5 * time.Millisecond) // reset takes 5ms
+
+	// enable all axes, normal mode
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL1, []byte{0x07})
+
+	// 400Hz rate
+	d.SetDataRate(LIS3DH_DATARATE_400_HZ)
+
+	// High res & BDU enabled
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL4, []byte{0x88})
+
+	// DRDY on INT1
+	d.bus.WriteRegister(uint8(d.Address), LIS3DH_REG_CTRL3, []byte{0x10})
```

Comment:
It seems like there is no support for interrupts yet? Why do you enable them?

Reply:
Removed